lib/torba/verify.rb

```diff
@@ -1,7 +1,14 @@
 require "torba"
 
-if STDOUT.tty?
-  Torba.pretty_errors { Torba.verify }
-else
-  Torba.verify
+# We purposely skip verification during Rake execution, because Rake is used to
+# run the Torba:RakeTask during the "assets:precompile" stage of deployment. In
+# this case it is OK that the Torbafile hasn't been packed yet, because we are
+# about to do so.
+#
+unless defined?(Rake::Task)
```

Comment:
The results of my experiments:

rails 4.2 (w/o spring):

command | defined?(Rake::Task) | $0
------------- | ---------------------------- | -------
bin/rake db:migrate | nil | "bin/rake"
bin/rake assets:precompile | nil | "bin/rake"
bundle exec rake db:migrate | "constant" | "/Users/nash/.rbenv/versions/2.2/bin/rake"
bundle exec rake assets:precompile | "constant" | "/Users/nash/.rbenv/versions/2.2/bin/rake"

rails 5.0.0.beta.1.1 (w/ srping)

command | defined?(Rake::Task) | $0
------------- | ---------------------------- | -------
bin/rake db:migrate | nil | "spring app    \| rails5 \| started 0 secs ago \| development mode"
bin/rake assets:precompile | nil | "spring app    \| rails5 \| started 0 secs ago \| development mode"
bin/rails db:migrate | nil | "bin/rails"
bin/rails assets:precompile | nil | "bin/rails"
bundle exec rake db:migrate | "constant" | "/Users/nash/.rbenv/versions/2.2/bin/rake"
bundle exec rake assets:precompile | "constant" | "/Users/nash/.rbenv/versions/2.2/bin/rake"

I assume 5x w/o spring is the same as 4.2.

You're correct, we can't rely neither on `Rake::Task`, nor on `$0`. Maybe use `ARGV`, but use whitelist approach instead? We only need assets related verification when "server" is run or `RAILS_ENV` is test.